flow/record/fieldtypes/__init__.py

```diff
@@ -640,4 +648,13 @@ class posix_path(pathlib.PurePosixPath, path):
 
 
 class windows_path(pathlib.PureWindowsPath, path):
-    pass
+    def __repr__(self) -> str:
+        s = str(self)
+        quote = "'"
+        if "'" in s:
+            if '"' in s:
+                s = s.replace("'", "\\'")
+            else:
+                quote = '"'
+
+        return f"{quote}{s}{quote}"
```

Comment:
It shouldn't? It uses the `__repr__` defined in the base `path` class here.

Reply:
the quotation handling only happens for windows_path though, it should also apply for posix_path or path in general?  There are also tests missing for posix_path quotation edge cases.

Reply:
hmm then I probably don't understand why you have to emulate repr for windows_path but not for posix_path. I thought it was to get a consistent single or doublequoted string.

Reply:
Because for Windows paths it will also escape the "\" to "\\" (or "\\\\" if you want to go by the correct notation, which will print as "\\"). IMO this is not wanted (you want to print a single backslash), so I didn't want to use repr() directly, hence the emulation of how repr() handles quotes in strings internally.

Reply:
gotcha, thanks for explaining!

Reply:
> For posix I just call `repr(str(self))` currently, so Python handles that for us with the call to `repr()`.

AH I missed the base `path` class changes to `__repr__` apparently.

LGTM